dts/bindings/serial/nordic,nrf-uart-common.yaml

```diff
@@ -43,3 +43,6 @@ properties:
       - 460800
       - 921600
       - 1000000
+      - 2000000
```

Comment:
shouldn't we introduce `nordic,nrf-uart-fast.yaml` or similar to differentiate between devices that have >1Mbaud capabilities? Take a look at this PR for reference: https://github.com/zephyrproject-rtos/zephyr/pull/79330

Reply:
Driver compilation will fail if not supported `current-speed` is set for an instance. Alternative approach would be to use separate compatible but this will trigger more work as currently `current-speed` is an enum in `nordic,nrf-uart-common.yaml` and that would need to be extracted because enums in bindings cannot be overwritten. Additionally, support for two compatibles would need be added in places like: https://github.com/zephyrproject-rtos/zephyr/blob/c3f979b7367f892ec4ec63b9ebb102810e910050/drivers/pinctrl/pinctrl_nrf.c#L45. I'm not sure what we gain apart from error detected in DT vs error detected during compilation.

Reply:
If we where completely true to the devicetree, if its the "same" underlying hardware block, and "fast" just means, "supports higher bitrate", we would add a flag/boolean to the common binding like:
```
uart@xxx {
        compatible = "nordic,nrf-uarte"; /* same hardware, targets same driver */
        fast; /* <this instance is fast */
};
```
the enum should contain a superset of all supported speeds, for all variations, including fast, the driver then validates `current-speed` based on if the `fast;` flag exists. DT really is not there to validate stuff like `current-speed`, `current-speed` is a zephyr-ism we use to set initial configs, DT enums are just a "base filter"

Reply:
Consider how many `nordic,nrf-comp` variations there would be if we had one compat if it supported low power mode, one if it supported some specific subset of voltage references etc. pure chaos :)

Reply:
`fast` property is currently redundant because fast instance can be determined based on clock frequency that drives it (16MHz - slow, >16MHz fast, currently 128M in nrf54l and 320M in nrf54h). Based on frequency that clocks the uart at compile time there is a build time check if `current-speed` can be achieved.

Reply:
Perfect, achieves the same thing in this case if the clock can not be selected at runtime :) Though a runtime check would also be needed if UART can be configured runtime :)

Reply:
runtime check is also there.